parser.y

```diff
@@ -5083,6 +5083,22 @@ CastType:
 		x.Collate = charset.CollationBin
 		$$ = x
 	}
+|	"FLOAT" FloatOpt
+	{
+		x := types.NewFieldType(mysql.TypeFloat)
+		fopt := $2.(*ast.FloatOpt)
+		if fopt.Flen >= 54 {
+			yylex.AppendError(ErrTooBigPrecision.GenWithStackByArgs(fopt.Flen,"CAST",53))
+		}
+		if fopt.Flen >= 25 {
```

Comment:
Maybe use `else if` here, though it doesn't really matter 🙃

Reply:
update.